### PR TITLE
{vis}[foss/2020b] IMOD v4.11.5

### PR DESCRIPTION
**Pull request by ThomasHoffmann77**
(created using `eb --new-pr`)

### CODE REVIEW

**Review by smoors — COMMENTED** [review]
@ThomasHoffmann77 why didn't you update the existing IMOD easyconfig, IMOD-4.7.15_RHEL6-64_CUDA6.0.eb, which uses the custom IMOD easyblock?

some things that seem missing compared to the existing easyconfig/easyblock:
-CUDA dependency
-the versionsuffix
-the subm alias: you recommend using the python subm script, shouldn't there be a Python dep then?

**Comment by verdurin** [comment]
@smoors Are you happy with this now?

**Comment by smoors** [comment]
I still don't understand why you did not use the custom IMOD easyblock?

File: easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+prebuildopts = local_exports
+
+# create some missing directories required for installation process:
+preinstallopts = 'mkdir %(installdir)s/{man/cat1,bin,autodoc,SystemTemplate,lib/imodplug,com,html,Plugins} -p &&'
+
+modextravars = {
+    'IMOD_DIR': "%(installdir)s",
+    'IMOD_PLUGIN_DIR': '%(installdir)s/lib/imodplug',
+    'IMOD_JAVADIR': '$JAVA_HOME',
+    'FOR_DISABLE_STACK_TRACE': '1',
```
Comment (smoors) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb:70]: shouldn't 'IMOD_QTLIBDIR' be added here?

File: easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+modloadmsg = 'Please set the environment variable $IMOD_CALIB_DIR if appropriate.'
+# Observed problems with alias on a tcl based module system. Use bin/subm python script instead.
+# modaliases = {
```
Comment (smoors) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb:74]: why did you comment it out? is there a downside to enabling this by default?

File: easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+toolchain = {'name': 'foss', 'version': '2020b'}
+toolchainopts = {'pic': True, 'openmp': True}
+
+sources = [SOURCE_TAR_GZ]
+checksums = ['5c50f1457134fe5113b95e0939d3181b948d211f25d9a05ad2b4ec0f23251c79']
+
+# mercurial repository
+# hg clone --debug http://bio3d.colorado.edu/imod/nightlyBuilds/IMOD
+# get lunch
```
Comment (smoors) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb:24]: I see what you did here :)

**Review by smoors — CHANGES_REQUESTED** [review]
@ThomasHoffmann77 ok, difference is that the easyblock does a binary install and this is a from source build.
I would still try to merge the 2 installation methods into the easyblock at some point, because the amount of customization is a bit much here.

File: easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb (hunk 1 above)
Comment (ThomasHoffmann77) [reply easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb:70]: in the binary installation IMOD_QTLIBDIR points to qtlib, which contains libQt5*, libtiff, libcufft, and others - all provided by dependency modules in the source build. Here the directory does not exist.
Thus I did not set the IMOD_QTLIBDIR variable.

File: easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb (hunk 2 above)
Comment (ThomasHoffmann77) [reply easybuild/easyconfigs/i/IMOD/IMOD-4.11.5-foss-2020b.eb:74]: I decided, not to enable the alias in order to make the EC more universal. As there is also a subm python script avialable, the alias is not mandatory.

**Comment by ThomasHoffmann77** [comment]
> @ThomasHoffmann77 ok, difference is that the easyblock does a binary install and this is a from source build.
> I would still try to merge the 2 installation methods into the easyblock at some point, because the amount of customization is a bit much here.
@smoors I agree. However, I have no experience/idea how to derive an easyblock from both, Binary and ConfigureMake. Do you have an example in mind?
How much customization would be OK? Some of them might be better placed in a patch, e.g. passing CFLAGS to the .pro files or creating missing directories during installation.

**Comment by smoors** [comment]
@ThomasHoffmann77 sorry for the long wait. if you can submit a test report for this PR I'll be happy to merge it.

**Comment by ThomasHoffmann77** [comment]
> @ThomasHoffmann77 sorry for the long wait. if you can submit a test report for this PR I'll be happy to merge it.

@smoors how should this test report look like?

**Comment by smoors** [comment]
you can run the following command for that:
`eb --from-pr=12499 -dfru`

**Comment by ThomasHoffmann77** [comment]
Test report by @thomashoffmann77
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
srv-eustermann-01 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7542 32-Core Processor, Python 2.7.5
See https://gist.github.com/f14f26e6f837acd24055be27d5c6ace4 for a full test report.

**Comment by ThomasHoffmann77** [comment]
Test report by @thomashoffmann77
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
srv-eustermann-01 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7542 32-Core Processor, Python 2.7.5
See https://gist.github.com/6037a8a10a6780c81f1ea8013d9eb617 for a full test report.

**Comment by ThomasHoffmann77** [comment]
Test report by @thomashoffmann77
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
srv-eustermann-01 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7542 32-Core Processor, Python 2.7.5
See https://gist.github.com/e10bd5b331a030d9a87ea3fa5c37625b for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @ThomasHoffmann77!

**Comment by boegel** [comment]
Some additional fixes for this in #13899...

**Pull request merged**